### DISCUSSION TITLE
first-time SSC import/.plr SSC inport

### DISCUSSION THREAD

**Discussion by antrobot1234** [Pryaxis/TShock#1738] [General]
**Is your feature request related to a problem? Please describe.**

i want to make my server SSC so one of my players isn't _tempted_ to cheat, but since we're well our way to getting hard mode, i don't want to wipe all progress.

**Describe the solution you'd like**

 I just want to read the player's information and set that to the SSC information for the first time they join. Then, they'll be forced to use the SSC version of that data. Alternatively, i'd like to import their plr files to the server and have them use that (assigning it to a certain username)

**Describe alternatives you've considered**

the only information i've seen about PLR to SSC conversion is for the reverse (SSC to PLR) and one really old post on the forums about a JAVA program that the poster never actually provided.

**Comment by hakusaro** [comment]
Couldn't you just use `/uploadssc playername`? You could turn on SSC, and then run that? Does that work?

**Comment by antrobot1234** [comment]
I looked at the help for uploadssc and it said it uploaded their player data from when they first joined. does that mean that i have to make everyone create new accounts for it to upload the data properly?

**Comment by hakusaro** [comment]
@antrobot1234 sorry, that's confusing. The message refers to the data when the user joined the server on the most recent connection. The system creates a snapshot of their data so that if it changes when they're playing on the server, `uploadssc` will always give them what they had when they connected (and not after playing). It doesn't store the data otherwise, so rejoining gives you a fresh snapshot. In other words, it doesn't refer to when their account was created or when they first joined the server globally, but the point in time when they first connected to that session.

**Comment by antrobot1234** [comment]
So, given i'm understanding this correctly, if i have each of my players make an account and join with their players then i can uploadssc for each of their accounts, close the server, set it to SSC mode, and re-open it and they should still have their data? 

also, two final notes. 

1:Would it work if they sent me their PLR file and i created an account for them, joined with their plr and logged into that account, uploaded the ssc, and repeated? (for people who aren't on during the shift)

 2:how do I check all the accounts currently registered on my server? (it's not required, but it would be useful to know)

**Comment by hakusaro** [comment]
1. They have to have an account. If they don't have an account, they either need one made or they should make one.
2. When they join, you can run `uploadssc playername` from the console after they've logged in.
3. You could probably do that flow, yes. You need to be able to switch users so you'd have to turn off UUID login.
4. Like, check which ones have SSC data?

**Comment by antrobot1234** [comment]
4: that, and to look at the list of all users

**Comment by hakusaro** [comment]
Your best bet is to use an sqlite viewer program to open the database and look at the character and user tables directly to see which accounts exist and which users have data. Other methods wouldn't be very fast.

**Comment by hakusaro** [comment]
I'm converting this to a discussion because I think we've covered your use case here. You can change the sorting on the right hand side of the screen to "chronological" from "most relevant" to preserve sorting order from this issue.
